# Two novel loci underlie natural differences in *Caenorhabditis elegans* macrocyclic lactone responses

Parasitic nematodes cause a massive worldwide burden to human health along with a loss of livestock and agriculture productivity. Anthelmintics have been widely successful in treating parasitic nematodes. However, resistance is increasing, and little is known about the molecular and genetic causes of resistance. The free-living roundworm *Caenorhabditis elegans* provides a tractable model to identify genes that underlie resistance. Unlike parasitic nematodes, *C. elegans* is easy to maintain in the laboratory, has a complete and well annotated genome, and has many genetic tools. Using a combination of wild isolates and a panel of recombinant inbred lines constructed from crosses of two genetically and phenotypically divergent strains, we identified three genomic regions on chromosome V that underlie natural differences in response to the macrocyclic lactone (ML) abamectin. One locus was identified previously and encodes an alpha subunit of a glutamate-gated chloride channel (*glc-1*). Here, we validated and narrowed two novel loci using near-isogenic lines. Additionally, we generated a list of prioritized candidate genes identified in *C. elegans* and in the parasite *Haemonchus contortus* by comparison to ML resistance loci. These genes could be unique to *C. elegans* or represent previously unidentified resistance genes shared with parasitic nematodes and should be evaluated using CRISPR-Cas9-mediated genome editing in the future. Our work highlights the advantages of using *C. elegans* as a model to better understand ML resistance in parasitic nematodes.

### Files
- **S1 File. Dose response phenotype data.** Results of the dose response for the genome-wide association high-throughput fitness assay
- **S2 File. Wild isolate phenotype data.** Residual phenotypic values for the 210 wild isolates in response to abamectin
- **S3 File. Association mapping results.** Genome-wide association mapping results for all four drug-response traits tested in the high-throughput fitness assay
- **S4 File. RIAIL phenotype data.** Residual phenotypic values for the 225 set 2 RIAILs in response to abamectin
- **S5 File. Linkage mapping results.** Linkage mapping LOD scores at 13,003 genomic markers for all four drug-response traits with the set 2 RIAILs 
- **S6 File. Summary of two-dimensional genome scan.** Summary of the scan2 object containing data from the two-dimensional genome scan with animal optical density (mean.EXT) in abamectin 
- **S7 File. Chromosome V variants.** Correlation values and annotations for all variants on chromosome V
- **S8 File. NIL sequence data.** VCF from the whole-genome sequencing for all the NILs in this study 
- **S9 File. NIL genotype data.** Simplified genotypes of the NILs in the study 
- **S10 File. CSSV phenotype data.** Raw pruned phenotypes for the high-throughput fitness assay with the chromosome V substitution strains
- **S11 File. Statistical significance for NIL assays.** Pairwise statistical significance for all strains and high-throughput assays 
- **S12 File. ChrV NIL breakup phenotype data.** Raw pruned phenotypes for the NILs used to break up the QTL interval on chromosome V 
- **S13 File. Candidate genes on chrV.** List of all genes in the chromosome VL and VC intervals, their functional descriptions and GO annotations, and if they have variation in CB4856 
- **S14 File. Set 1 RIAIL phenotype data.** Residual phenotypic values for the 121 set 1 RIAILs in response to abamectin
- **S15 File. Mediation estimates for chrV QTL.** Mediation estimates for the chromosome VL and VC QTL
- **S16 File. NILs to narrow VC QTL.** Raw pruned phenotypes for the chromosome VC NIL high-throughput fitness assay 
- **S17 File. *lgc-54* mutant phenotypes.** Raw pruned phenotypes for the the *lgc-54* deletion strains high-throughput fitness assay
- **S18 File. Orthologous genes in *H. contortus*.** Coordinates and orthologous relationships for *H. contortus* QTL
